### PR TITLE
Homepage screen

### PR DESCRIPTION
**Pull request by shrey27**
Added Features
- Homepage screen
- Dark theme
- mobile view responsive
- routing configuration

ss_1

ss_2

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    grid-template-areas: 'sidebar main';
+  }
+  .main {
+    min-height: 80vh;
+  }
+}
+
+@media screen and (min-width: 1650px) {
+  body {
+    width: 1650px;
```
Comment (Parul1999) [inline src/App.css]: Can use relative units.

File: src/App.css [hunk 2]
```diff
@@ -0,0 +1,137 @@
+@import url('https://apolloui-neog.netlify.app/utility/common.css');
+@import url('https://apolloui-neog.netlify.app/components/drawer/drawer.css');
+@import url('https://apolloui-neog.netlify.app/components/modal/modal.css');
+@import url('https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css');
+
+:root {
+  --secondary: rgb(255, 239, 0);
+  --tertiary: rgb(0, 49, 83);
+  --tertiary--light: #0073cf;
+  --background: #e9edf1;
```
Comment (Parul1999) [inline src/App.css:10]: For consistency, can either use rgba or hash for colors.

File: src/App.js [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+import { Navbar, Footer } from './frontend/components';
+import { useTheme } from './frontend/context';
+
 function App() {
-  return <div>Social Media App</div>;
+  const { theme } = useTheme();
+  return (
+    <div data-theme={theme}>
+      <Navbar />
+      {availableRoutes}
```
Comment (Parul1999) [inline src/App.js:10]: Nice code-structure.

File: src/frontend/pages/homepage/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import './homepage.css';
+import { Sidebar } from '../../components';
+import { posts } from '../../utility/constants';
+/**
```
Comment (Parul1999) [inline src/frontend/pages/homepage/index.js]: Remove the commented code

File: src/frontend/components/sidebar/index.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Fragment } from 'react';
+import './sidebar.css';
+import { NavLink } from 'react-router-dom';
+import { navlinks } from '../../utility/constants';
+
+// const backgroundStyle = ({ isActive }) => {
```
Comment (Parul1999) [inline src/frontend/components/sidebar/index.js]: Remove the commented code

File: src/frontend/components/navbar/navbar.css [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+.header__nav__image {
+  width: 6rem;
+  height: auto;
+  margin: 0;
+  object-fit: contain;
+}
+.header__nav__brand {
+  font-size: 1.35rem;
+  font-family: 'Allerta Stencil', sans-serif;
+  color: var(--tertiary);
```
Comment (Parul1999) [inline src/frontend/components/navbar/navbar.css:36]: Use variables for font family

File: src/frontend/components/navbar/navbar.css [hunk 7]
```diff
@@ -0,0 +1,132 @@
+@import '../../../App.css';
+@import url('https://fonts.googleapis.com/css2?family=Allerta+Stencil&display=swap');
+
+.navbar {
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-template-areas:
+    'start end'
+    'middle middle';
```
Comment (Parul1999) [inline src/frontend/components/navbar/navbar.css]: Instead of start middle end , the names could be more descriptive e.g nav-start

**Review by rohit8686 — APPROVED** [review]
- Clean code and UI looks great ! 
- Looks good to merge

**Comment by shrey27** [comment]
@Parul1999 i will resolve all the issues before merging

**Pull request merged**